### PR TITLE
Refactor terraform to stages, updating README with simplfied setups.

### PR DESCRIPTION
**Pull request by jonchenn**
This PR updates the terraform structure to use "stages" like in Foundation Toolkit fabrics: https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/fast/stages/00-bootstrap/main.tf.
 
It bundles the following changes:
- Refactor terraform structure to use stages, starting with foundation, gke and cloudrun.
- Adding setup_all.sh and e2e test script
- Update root README to cover the GCP setup steps, with simplified setup_all script and optional manual steps.

Please see this branch of how it looks like once merged: https://github.com/GoogleCloudPlatform/solutions-template/tree/e2e-test

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,44 +7,95 @@
 
 Solutions Template is a boilerplate template for building repeatable
 solutions with the best practices in architecture on Google Cloud, including GKE
-clusters, Test Automation, CI/CD, as well as development process.
+clusters, CloudRun, Test Automation, CI/CD, as well as development process.
```
Comment (lavjain) [inline README.md]: Minor: Cloud Run (2 words)

File: README.md [hunk 2]
```diff
@@ -7,44 +7,95 @@
...
-Install Cookiecutter ([Github](https://github.com/cookiecutter/cookiecutter)):
+We recommend the following resources to get familiar with Google Cloud and microservices.
+
+- What is Microservice Architecture: https://cloud.google.com/learn/what-is-microservices-architecture
+- Kubernetes:
+  - Learn about the basics of Kubernetes: https://kubernetes.io/docs/concepts/overview/
+  - Google Kubernetes Engine (GKE) overview: https://cloud.google.com/kubernetes-engine/docs/concepts/kubernetes-engine-overview
+  - Skaffold, a command line tool that facilitates continuous development for container based & Kubernetes applications: https://skaffold.dev/docs/
+- CloudRun:
+  - Serverless container deployment and execution with CloudRun: https://cloud.google.com/run/docs/overview/what-is-cloud-run
```
Comment (lavjain) [inline README.md]: The hyper links can be embedded:
 
- What is [Microservice Architecture](https://cloud.google.com/learn/what-is-microservices-architecture)
- Kubernetes:
  - Learn about the basics of [Kubernetes](https://kubernetes.io/docs/concepts/overview/)
  - [Google Kubernetes Engine (GKE)](https://cloud.google.com/kubernetes-engine/docs/concepts/kubernetes-engine-overview) overview
  - [Skaffold](https://skaffold.dev/docs/), a command line tool that facilitates continuous development for container based & Kubernetes applications
- Cloud Run:
  - Serverless container deployment and execution with [Cloud Run](https://cloud.google.com/run/docs/overview/what-is-cloud-run)

File: README.md (hunk 1 above)
Comment (jonchenn) [reply README.md]: Resolved.

File: README.md (hunk 2 above)
Comment (jonchenn) [reply README.md]: Resolved.

**Comment by jonchenn** [comment]
@lavjain Updated, please review.

**Review by lavjain — APPROVED** [review]
Tested GKE cluster (auto) and Cloud Run (manual)

**Pull request merged**